src/filesystem.ts

```diff
@@ -45,7 +45,7 @@ export const path: typeof NodePath = Platform.isDesktopApp ? window.require('nod
 export const url: typeof NodeUrl = Platform.isDesktopApp ? window.require('node:url') : null;
 export const zlib: typeof NodeZlib = Platform.isDesktopApp ? window.require('node:zlib') : null;
 
-export function nodeBufferToArrayBuffer(buffer: Buffer, offset = 0, length = buffer.byteLength): ArrayBuffer {
+export function nodeBufferToArrayBuffer(buffer: Buffer<ArrayBuffer>, offset = 0, length = buffer.byteLength - offset): ArrayBuffer {
```

Comment:
Is this a bug in the current implementation? Did you catch it causing a problem, or did you find this through inspection?